osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/AttributeAddressHelperService.java

```diff
@@ -0,0 +1,248 @@
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.joda.time.DateTime;
+import org.openmuc.jdlms.AttributeAddress;
+import org.openmuc.jdlms.ObisCode;
+import org.openmuc.jdlms.SelectiveAccessDescription;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.PeriodTypeDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service(value = "attributeAddressHelperService")
+public class AttributeAddressHelperService {
+
+    private static final int CLASS_ID_PROFILE_GENERIC = 7;
+    private static final ObisCode OBIS_CODE_INTERVAL_BILLING = new ObisCode("1.0.99.1.0.255");
+    private static final ObisCode OBIS_CODE_DAILY_BILLING = new ObisCode("1.0.99.2.0.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_BILLING = new ObisCode("0.0.98.1.0.255");
+    private static final byte ATTRIBUTE_ID_BUFFER = 2;
+    private static final byte ATTRIBUTE_ID_SCALER_UNIT = 3;
+    private static final ObisCode OBIS_CODE_INTERVAL_IMPORT_SCALER_UNIT = new ObisCode("1.0.1.8.0.255");
+    private static final ObisCode OBIS_CODE_INTERVAL_EXPORT_SCALER_UNIT = new ObisCode("1.0.2.8.0.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_DAILY_IMPORT_RATE_1_SCALER_UNIT = new ObisCode("1.0.1.8.1.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_DAILY_IMPORT_RATE_2_SCALER_UNIT = new ObisCode("1.0.1.8.2.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_DAILY_EXPORT_RATE_1_SCALER_UNIT = new ObisCode("1.0.2.8.1.255");
+    private static final ObisCode OBIS_CODE_MONTHLY_DAILY_EXPORT_RATE_2_SCALER_UNIT = new ObisCode("1.0.2.8.2.255");
+
+    private static final int CLASS_ID_REGISTER = 3;
+
+    private static final byte[] OBIS_BYTES_ACTIVE_ENERGY_IMPORT_RATE_1 = new byte[] { 1, 0, 1, 8, 1, (byte) 255 };
+    private static final byte[] OBIS_BYTES_ACTIVE_ENERGY_IMPORT_RATE_2 = new byte[] { 1, 0, 1, 8, 2, (byte) 255 };
+    private static final byte[] OBIS_BYTES_ACTIVE_ENERGY_EXPORT_RATE_1 = new byte[] { 1, 0, 2, 8, 1, (byte) 255 };
+    private static final byte[] OBIS_BYTES_ACTIVE_ENERGY_EXPORT_RATE_2 = new byte[] { 1, 0, 2, 8, 2, (byte) 255 };
+    private static final byte ATTRIBUTE_ID_VALUE = 2;
+
+    private static final int ACCESS_SELECTOR_RANGE_DESCRIPTOR = 1;
+
+    private final DlmsHelperService dlmsHelperService;
+
+    @Autowired
+    public AttributeAddressHelperService(final DlmsHelperService dlmsHelperService) {
+        this.dlmsHelperService = dlmsHelperService;
+    }
+
+    public AttributeAddress[] getProfileBufferAndScalerUnitForPeriodicMeterReads(final PeriodTypeDto periodType,
+            final DateTime beginDateTime, final DateTime endDateTime, final boolean isSelectingValuesSupported)
```

Comment:
no need to repeat the 'DateTime' type in field/parameter/variable names. I would use `DateTime from` and `DateTime to` because it's a common convention to mark periods of time that way, but `begin` and `end` is also clear.

Reply:
Aangepast